src/spootnik/reporter.clj

```diff
@@ -55,29 +59,36 @@
                                             (s/optional-key :jmx)      report
                                             }}
      (s/optional-key :riemann) {:host                      s/Str
+                                (s/optional-key :port)     s/Num
+                                (s/optional-key :protocol) s/Str
                                 (s/optional-key :batch)    s/Num
-                                (s/optional-key :defaults) s/Any}}))
+                                (s/optional-key :defaults) s/Any
+                                (s/optional-key :ssl)      rssl}}))
 
 (def config-validator
   (s/validator config-schema))
 
 (defn riemann-client
   "To keep dependency conflicts, let's use RiemannClient directly."
-  [{:keys [host batch] :as opts}]
-  (let [port (or (:port opts) 5555)
-        tags (set (:tags opts))
-        tcp? (if-let [p (some-> opts :protocol keyword #{:tcp :udp})]
-               (= :tcp p)
-               true)]
-    (try
-      (let [client (if tcp?
-                     (com.aphyr.riemann.client.RiemannClient/tcp host (int port))
-                     (com.aphyr.riemann.client.RiemannClient/udp host (int port)))]
-        (if batch
-          (com.aphyr.riemann.client.RiemannBatchClient. client (int batch))
-          client))
-      (catch Exception e
-        (error e "cannot connect to riemann")))))
+  [{:keys [host port protocol batch ssl],
+    :or {port 5555, protocol "tcp"}}]
+  (try
+    (let [client (if (= protocol "tcp")
+                   (if ssl
+                     (RiemannClient/wrap
+                       (doto (new TcpTransport host (int port))
+                         (-> .-sslContext
+                             (.set (SSL/sslContext
+                                     (:pkey ssl)
+                                     (:cert ssl)
+                                     (:authority ssl))))))
+                     (RiemannClient/tcp host (int port)))
+                   (RiemannClient/udp host (int port)))]
```

Comment:
> you could switch to a cond

you mean literally `protocol "ssl" :ssl {...}` or just reusing the `TcpTransport` creation for plain tcp too?